homeassistant/components/config_entry_example.py

```diff
@@ -0,0 +1,102 @@
+"""Example component to show how config entries work."""
```

Comment:
I'm not sure if that should be inside components or call like `demo_component.py` and hold also other examples

Reply:
I will actually remove this component the moment we have 2/3 real implementations. I thought it would be a good example now for people to try it out themselves.

In the future I'll remove this component and add it to the demo platforms instead. But for that I would first need platform support for config entries (and this PR is already too big).